organization-bootstrap/business-units/main.tf

```diff
@@ -0,0 +1,162 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+#                        Terraform top-level resources                        #
+###############################################################################
+
+# Terraform project
+
+module "project-tf" {
+  source          = "terraform-google-modules/project-factory/google//modules/fabric-project"
+  version         = "3.2.0"
+  parent          = var.root_node
+  billing_account = var.billing_account_id
+  prefix          = var.prefix
+  name            = "terraform"
+  lien_reason     = "terraform"
+  owners          = var.terraform_owners
+  activate_apis   = var.project_services
+}
+
+# Bootstrap Terraform state GCS bucket
+
+module "gcs-tf-bootstrap" {
+  source     = "terraform-google-modules/cloud-storage/google"
+  version    = "1.0.0"
+  project_id = module.project-tf.project_id
+  prefix     = "${var.prefix}-tf"
+  names      = ["tf-bootstrap"]
+  location   = var.gcs_location
+}
+
+###############################################################################
+#                              Business units                                 #
+###############################################################################
+
+# Business unit 1
+
+module "business-unit-1-folders-tree" {
+  source = "./modules/business-unit-folders-tree"
+
+  billing_account_id            = var.billing_account_id
+  tf_project_id                 = module.project-tf.project_id
+  top_level_folder_name         = var.business_unit_1_name
+  second_level_folders_names    = var.business_unit_1_envs
+  generate_service_account_keys = var.generate_service_account_keys
+  gcs_location                  = var.gcs_location
+  organization_id               = var.organization_id
```

Comment:
I don't really like the way service accounts and bucket are managed implicitly by the module. It removes the documentation aspect which is really really important, and makes them more fragile. It also does not really match the actual use cases, where you have per-environment service accounts and buckets which are then used across all second-level folders.

Can we rework this so that the internal module is as lightweight as possible, just a simple wrapper to manage the folder hierarchy, and all the rest happens explicitly at the top level?